hail/src/main/scala/is/hail/variant/MatrixTable.scala

```diff
@@ -442,28 +442,14 @@ class MatrixTable(val hc: HailContext, val ast: MatrixIR) {
     (r: Row) => Row.fromSeq(queriers.map(_ (r)))
   }
 
-  def keyRowsBy(keys: java.util.ArrayList[String], partitionKeys: java.util.ArrayList[String]): MatrixTable =
-    keyRowsBy(keys.asScala.toArray, partitionKeys.asScala.toArray)
+  def keyRowsBy(keys: java.util.ArrayList[String]): MatrixTable =
+    keyRowsBy(keys.asScala.toFastIndexedSeq, false)
 
-  def keyRowsBy(keys: Array[String], partitionKeys: Array[String]): MatrixTable = {
-    require(keys.nonEmpty)
-    require(partitionKeys.nonEmpty)
-    val rowFields = rowType.fieldNames.toSet
-    assert(keys.forall(rowFields.contains), s"${ keys.filter(k => !rowFields.contains(k)).mkString(", ") }")
-    assert(partitionKeys.length <= keys.length)
-    assert(keys.zip(partitionKeys).forall { case (k, pk) => k == pk })
+  def keyRowsBy(keys: java.util.ArrayList[String], isSorted: Boolean): MatrixTable =
+    keyRowsBy(keys.asScala.toFastIndexedSeq, isSorted)
 
-    val newMatrixType = matrixType.copy(rowKey = keys)
-
-    copyMT(matrixType = newMatrixType, rvd = rvd.changeKey(keys))
-  }
-
-  def keyColsBy(keys: java.util.ArrayList[String]): MatrixTable = keyColsBy(keys.asScala: _*)
-
-  def keyColsBy(keys: String*): MatrixTable = {
```

Comment:
What happened to keyColsBy?

Reply:
It wasn't being used. Changing column keys just goes through `selectCols`.